File: python/S1_BASIC/L4_BASICMATHS/armstrong.py
class Solution:
    def __init__(self,num):
        self.num=num
    def digitcount(self,n):
        count=0
        while n>0:
            count+=1
            n=n//10
        return count

    def armstrong(self):
        temp=self.num
        count=self.digitcount(temp)
        sum=0
        while temp>0:
            d=temp%10
            sum+=d**count
            temp//=10
        return sum==self.num
if __name__=="__main__":
    n=int(input("enter number"))
    s=Solution(n)
    print(s.armstrong())

File: JAVA/S1_BASIC/L4_BASICMATHS/armstrong.java

class solution{
    int countdigit(int num){
        int count=0;
        int n=num;
        while(n>0){
            count++;
            n/=10;
        }
        return count;
        
    }
    int armstrong(int num){
        int n=countdigit(num);
        int n2=num;
        int sum=0;
        while(n2>0){
            sum+=Math.pow((n2%10),n);
            n2/=10;

        }
        if(sum==num)return 1;
        return 0;
    }
}
public class armstrong{
    public static void main(String[] args){
        solution s=new solution();
        System.out.println(s.armstrong(103));
    }
}